server/models/annotation.py

```diff
@@ -0,0 +1,534 @@
+        if properyName != 'type'
+    }
+    baseRectangleShapePatternProperties.update(baseShapePatternProperties)
+    rectangleShapeSchema = {
+        'allOf': [
+            baseRectangleShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {
+                        'type': 'string',
+                        'enum': ['rectangle']
+                    },
+                },
+                'required': ['type'],
+                'patternProperties': baseRectangleShapePatternProperties,
+                'additionalProperties': False
+            }
+        ]
+    }
+    rectangleGridShapeSchema = {
+        'allOf': [
+            baseRectangleShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {
+                        'type': 'string',
+                        'enum': ['rectanglegrid']
+                    },
+                    'widthSubdivisions': {
+                        'type': 'integer',
+                        'minimum': 1
+                    },
+                    'heightSubdivisions': {
+                        'type': 'integer',
+                        'minimum': 1
+                    },
+                },
+                'required': ['type', 'widthSubdivisions', 'heightSubdivisions'],
+                'patternProperties': baseRectangleShapePatternProperties,
+                'additionalProperties': False,
+            }
+        ]
+    }
+
+    annotationSchema = {
+        '$schema': 'http://json-schema.org/schema#',
+        'id': '/girder/plugins/large_image/models/annotation',
+        'type': 'object',
+        'properties': {
+            'name': {
+                'type': 'string',
+                # TODO: Disallow empty?
+                'minLength': 1,
+            },
+            'description': {'type': 'string'},
+            'attributes': {
+                'type': 'object',
+                'additionalProperties': True,
+                'title': 'Image Attributes',
+                'description': 'Subjective things that apply to the entire '
+                               'image.'
+            },
+            'elements': {
+                'type': 'array',
+                'items': {
+                    # Shape subtypes are mutually exclusive, so for
+                    #  efficiency. don't use 'oneOf'
+                    'anyOf': [
+                        # If we include the baseShapeSchema, then shapes that
+                        # are as-yet invented can be included.
+                        # baseShapeSchema,
+                        arrowShapeSchema,
+                        circleShapeSchema,
+                        pointShapeSchema,
+                        polylineShapeSchema,
+                        rectangleShapeSchema,
+                        rectangleGridShapeSchema,
+                    ]
```

Comment:
Good thought - major/minor axis is the other common way people would deal with ellipses

________________________________
From: Deepak Roy Chittajallu <notifications@github.com>
Sent: Tuesday, April 12, 2016 12:22:59 PM
To: DigitalSlideArchive/large_image
Cc: Cooper, Lee
Subject: Re: [DigitalSlideArchive/large_image] Annotations RESTful interface (#50)


In server/models/annotation.py<https://github.com/DigitalSlideArchive/large_image/pull/50#discussion_r59406502>:

> +            'elements': {
> +                'type': 'array',
> +                'items': {
> +                    # Shape subtypes are mutually exclusive, so for
> +                    #  efficiency. don't use 'oneOf'
> +                    'anyOf': [
> +                        # If we include the baseShapeSchema, then shapes that
> +                        # are as-yet invented can be included.
> +                        # baseShapeSchema,
> +                        arrowShapeSchema,
> +                        circleShapeSchema,
> +                        pointShapeSchema,
> +                        polylineShapeSchema,
> +                        rectangleShapeSchema,
> +                        rectangleGridShapeSchema,
> +                    ]


Yes i would stick with center, width, height, and rotation similar to rectangle as they are intuitive and given that other code will have to provide these things while creating/editing the annotation.

I was thinking maybe we should use major axis length and minor axis length instead of width and height, respectively but come to think of it leaving them as width (for major axis length) and height (minor axis length) seems more intuitive.

-
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/DigitalSlideArchive/large_image/pull/50/files/149c5ffc59cebe59cea00b1f051dbecc2db4c549#r59406502>

________________________________

This e-mail message (including any attachments) is for the sole use of
the intended recipient(s) and may contain confidential and privileged
information. If the reader of this message is not the intended
recipient, you are hereby notified that any dissemination, distribution
or copying of this message (including any attachments) is strictly
prohibited.

If you have received this message in error, please contact
the sender by reply e-mail message and destroy all copies of the
original message (including attachments).